mrmustard/lab_dev/circuit_components.py

```diff
@@ -47,40 +47,55 @@ class CircuitComponent:
 
     def __init__(
         self,
-        name: str,
+        name: Optional[str] = None,
         representation: Optional[Representation] = None,
         modes_out_bra: Optional[Sequence[int]] = None,
         modes_in_bra: Optional[Sequence[int]] = None,
         modes_out_ket: Optional[Sequence[int]] = None,
         modes_in_ket: Optional[Sequence[int]] = None,
     ) -> None:
-        self._name = name
+        self._name = name or ""
         self._wires = Wires(modes_out_bra, modes_in_bra, modes_out_ket, modes_in_ket)
         self._parameter_set = ParameterSet()
         self._representation = representation
 
     @classmethod
-    def from_attributes(
-        cls,
-        name: str,
-        representation: Representation,
-        wires: Wires,
-    ):
+    def _from_attributes(
+        cls, name: str, representation: Representation, wires: Wires
+    ) -> CircuitComponent:
         r"""
         Initializes a circuit component from its attributes (a name, a ``Wires``,
         and a ``Representation``).
 
+        If the Method Resolution Order (MRO) of ``cls`` contains one between ``Ket``, ``DM``,
+        ``Unitary``, and ``Channel``, then the returned component is of that type. Otherwise,
+        it is of type ``CircuitComponent``.
+
+        This function needs to be used with caution, as it does not check that the attributes
+        provided are consistent with the type of the returned component. If used improperly it
+        may initialize, e.g., ``Ket``s with both input and output wires or ``Unitary``s with
```

Comment:
Just a small question, "If used improperly it may initialize" means we can always use this function even though the information given are not proper right?

Reply:
We could do something like:
```
state = Coherent([0], x=1)
unitary = BSgate([0, 1], r=0.1)

Ket._from_state(name="", representation=state.representation, wires=unitary.wires)
```
This would return an object which has the rep of a state and the wires of a unitary.

That's why it needs to be used with caution, and why it should not be shown to the users (with the underscore, it will not appear in the docs, so only devs can see it)